mlrun/utils/helpers.py

```diff
@@ -261,7 +261,11 @@ def normalize_name(name):
     # TODO: Must match
     # [a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?
     name = re.sub(r"\s+", "-", name)
-    name = name.replace("_", "-")
+    if "_" in name:
+        logger.warning(
+            "Replacing underscores (_) with dashes (-). Names with underscore will be deprecated, use dashes instead."
+        )
```

Comment:
I would change the order, so it will pop better when user looking at the logs. When you first state that you are replacing it seems like some enrichment has been executed, rather than a notice to stop using underscore.

Reply:
good point